src/main/java/seedu/address/MainApp.java

```diff
@@ -90,7 +92,19 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
             initialData = new AddressBook();
         }
 
-        return new ModelManager(initialData, userPrefs);
+        // update fees upon initialisation
+        Model model = new ModelManager(initialData, userPrefs);
+        FeesCalculator feesCalculator = new FeesCalculator(model.getLastUpdatedDate(), LocalDateTime.now());
+        model = feesCalculator.updateAllLessonOutstandingFees(model);
+
+        try {
+            storage.saveAddressBook(model.getAddressBook());
+        } catch (IOException io) {
+            logger.warning("Problem from saving TAB's last updated date. The data will not be updated"
+                    + "Please restart TAB.");
```

Comment:
```suggestion
            logger.warning("Encountered an error while saving TAB's last updated date. The data will not be updated."
                    + " Please restart TAB.");
```

Reply:
When will user encounter this error?

Reply:
There shouldn't be a message to restart TAB in the log I think

Reply:
when there is something wrong when saving address book, no way to replicate this also it is just a possible error that might happen

Reply:
There is no where else to put this error though @Chesterwongz